### ISSUE TITLE
tabclose works on pinned tabs

### ISSUE THREAD

**Issue by MisterVulcan** [cmcaine/tridactyl#344]
`:tabclose` and, by extension, `d` hotkey closes pinned tabs while `CTRL+W` does not, 1.8.2pre701 beta FF 58.0.2. Replicated in Linux and Windows.

**Comment by cmcaine** [comment]
Issues can be feature requests too.

I'd not noticed that CW doesn't close pinned tabs. I'm a bit reluctant to add this until we have feedback on the commandline (when CSS is hidden it's not obvious you're on a pinned tab), though I suppose in the long run we should be consistent with Firefox.

**Comment by IsaacKhor** [comment]
I’m actually happy with default behaviour, but if we’re aiming for consistency, I guess `:tabclose` on pinned tabs should issue a warning and abort; pass in `-f —force` to override

**Comment by bovine3dom** [comment]
Please +1 this issue if you want `tabclose` to have to be forced on pinned tabs. I'm assuming that most people don't due to the lack of traffic this is getting.

**Comment by glacambre** [comment]
@MisterVulcan Please note that if this is a feature you really want, you can implement it yourself. Just add `jsb tri.excmds.tabclose = $TABCLOSE_FROM_PR_721` to your tridactylrc and it'll do what you want it to do.

**Comment by bovine3dom** [comment]
You'd need to squash all the new lines and add semicolons where required : )

**Comment by notJerl** [comment]
Will that work if I use synced settings?

**Comment by glacambre** [comment]
@notJerl Yes it should. If you don't want to use a tridactylrc you could use an autocommand instead, e.g. `autocmd TriStart .* jsb tri.excmds.tabclose = $TABCLOSE_FROM_PR_721`.

**Comment by bew** [comment]
With the help of nice people on matrix, I got this working:
```js
jsb (async () => {
  const tab = await tri.webext.activeTab();
  alert(tab.id);
  if (!tab.pinned) {
    browser.tabs.remove(tab.id)
  }
  else {
    tri.excmds.fillcmdline_tmp(2000, "# tab is pinned, cannot close!")
  }
})()
```

And as a oneliner command `tabclose_safe`:
`command tabclose_safe jsb (async () => { const tab = await tri.webext.activeTab(); alert(tab.id); if (!tab.pinned) { browser.tabs.remove(tab.id) } else { tri.excmds.fillcmdline_tmp(2000, "# tab is pinned, cannot close!") } })()`
